.circleci/config.yml

```diff
@@ -172,11 +185,23 @@ jobs:
     docker:
       - image: circleci/node:16
     steps:
+      - update-npm
       - show-npm-version
       - provision-and-package-solcjs
       - provision-hardhat-with-packaged-solcjs
       - run:
-          name: Run hardhat-core test suite with its default solc
+          name: Restore the default solc binary expected by Hardhat
+          command: |
+            # Hardhat downloader tests are hard-coded to expect the version that comes with the solc-js.
+            # We forced latest solc-js but we still want the default binary with it.
+            hardhat_default_solc_version=$(jq --raw-output '.dependencies.solc' hardhat/packages/hardhat-core/package.json)
+            mkdir hardhat-default-solc/
+            pushd hardhat-default-solc/
+            npm install "solc@${hardhat_default_solc_version}"
+            popd
+            ln -sf ../../../hardhat-default-solc/node_modules/solc/soljson.js hardhat/node_modules/solc/soljson.js
```

Comment:
This part is needed for the downloader tests to pass. Otherwise when the original import problem is resolved they still fail like in [5174](https://app.circleci.com/pipelines/github/ethereum/solc-js/1175/workflows/1203dc75-62d7-435b-b66b-04008e693202/jobs/5174):

```
  1) Compiler downloader
       Downloaded compiler verification
         Shouldn't do anything if the compiler is fine:
     Error: ENOENT: no such file or directory, unlink '/tmp/hardhat-tests-compiler-downloader/wasm/list.json'
  

  2) Compiler downloader
       getDownloadedCompilerPath
         If the compiler already existed
           Should return it if it's passes the verification:
     Error: ENOENT: no such file or directory, stat '/home/circleci/project/hardhat/node_modules/solc/soljson.js'
```

The reason seems to be that while Hardhat ignores the binary from the package at runtime, [it does expect it to be there in tests](https://github.com/NomicFoundation/hardhat/blob/hardhat%402.9.5/packages/hardhat-core/test/internal/solidity/compiler/downloader.ts#L59). And it [assumes it to be specifically 0.7.3](https://github.com/NomicFoundation/hardhat/blob/hardhat%402.9.5/packages/hardhat-core/test/internal/solidity/compiler/downloader.ts#L29-L41). I think that the first test that fails tries to verify the binary and verification routine is deletes it if it's not the version it expects, which is the case here.

I think that the only reason it wasn't failing before was that the solc-js injection into dependencies was not being performed correctly.

Reply:
Given all this craziness I think that the simple test in https://github.com/ethereum/solc-js/pull/625#discussion_r876059739 will probably be the most reliable indicator in the future :-).

Reply:
Yeah, for simple crashes like this one. For more tricky cases without an outright crash this one might still be useful.